luwak/src/main/java/uk/co/flax/luwak/BatchMonitor.java

```diff
@@ -0,0 +1,333 @@
+package uk.co.flax.luwak;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import uk.co.flax.luwak.Monitor.QueryCacheStats;
+import uk.co.flax.luwak.presearcher.PresearcherMatches;
+
+/**
+ * A Monitor contains a set of MonitorQuery objects, and runs them against
+ * passed-in InputDocuments.
+ */
+public class BatchMonitor implements Closeable {
+
+    private Monitor monitor;
+    private long commitTimeout;
+    private long lastCommitTime = 0;
+    private volatile ScheduledFuture<?> commitJob = null;
+    private final ScheduledExecutorService commitExecutor;
+
+    private Map<String, Indexable> pendingUpdates = new ConcurrentHashMap<>();
+    private Queue<String> pendingDeletes = new ConcurrentLinkedQueue<>();
+
+    /**
+     * Create a new BatchMonitor instance, that writes to the supplied monitor
+     *
+     * Note that when the BatchMonitor is closed, the embedded Monitor
+     * (and its all components) will also be closed.
+     *
+     * @param monitor the underlying monitor to write to
+     * @param configuration the MonitorConfiguration
+     * @throws IOException on IO errors
+     */
+    public BatchMonitor(Monitor monitor, BatchQueryIndexConfiguration configuration) throws IOException {
+        this.monitor = monitor;
+        this.commitTimeout = configuration.getCommitTimeout();
+        this.commitExecutor = Executors.newSingleThreadScheduledExecutor();
+    }
+
+    /**
+     * Create a new Monitor instance, using a RAMDirectory and the default configuration
+
+     * @param monitor the underlying monitor to write to
+     * @throws IOException on IO errors
+     */
+    public BatchMonitor(Monitor monitor) throws IOException {
+        this.monitor = monitor;
+        this.commitTimeout = 0;
+        this.commitExecutor = Executors.newSingleThreadScheduledExecutor();
+    }
+
+    /**
+     * Register a {@link QueryIndexUpdateListener} that will be notified whenever changes
+     * are made to the Monitor's queryindex
+     *
+     * @param listener listener to register
+     */
+    public void addQueryIndexUpdateListener(QueryIndexUpdateListener listener) {
+        monitor.addQueryIndexUpdateListener(listener);
+    }
+
```

Comment:
I don't think we need all these delegating methods?  Clients can just pass their Monitor to the this object, but hold a reference to the original Monitor for querying/selecting.